src/preamp.jl

```diff
@@ -58,10 +80,16 @@ function GenericPreAmp(preamp_config::PropDict)
     GenericPreAmp(
         τ_decay=T(preamp_config.t_decay)*u"μs",
         τ_rise=T(preamp_config.t_rise)*u"ns",
-        offset = T(preamp_config.offset)u"keV",
-        max_e = T(preamp_config.max_e)u"keV",
-        noise_σ = haskey(preamp_config, :noise_sigma) ? T(preamp_config.noise_sigma)u"keV" : 0u"keV"
+        # optional, if not given usually means NoiseFromData is used -> make a better system for managing this!
+        offset_keV = haskey(preamp_config, :offset_in_keV) ? T(preamp_config.offset_in_keV)u"keV" : 0u"keV",
+        offset_ADC = haskey(preamp_config, :offset_in_ADC) ? T(preamp_config.offset_in_ADC) : 0,
+        # max_e = T(preamp_config.max_e)u"keV",
+        gain = preamp_config.gain/u"eV",
```

Comment:
The tests fail here because apparently, `preamp_config.gain` is missing.

Reply:
Updated the tests, thanks.